docs/src/understand/thin-edge-json.md

```diff
@@ -114,44 +169,59 @@ and hence must not be used as measurement keys:
+*Events* are notifications that something happened on the device, its environment, the domain application or the software system.
+For instance:
+- a door has been closed
+- a process started
+- a user has started an ssh session
 
-Here is the complete list of topics reserved by `thin-edge.io` for its internal working:
+```sh te2mqtt
+tedge mqtt pub /tedge/events/login '{
+  "text": "A user just logged in",     // 'text' message of that event
+  "time": "2021-01-01T05:30:45+00:00"  // optional 'timestamp' of that event
+  "someOtherCustomFragment": {         // free 'custom fragments'
+    "nested": {
+      "value": "extra info"
+    }
+  }
+}'
+```
 
-| Topic | Description |
-| --- | --- |
-| `tedge/` | Reserved root topic of `thin-edge.io` |
-| `tedge/measurements` | Topic to publish measurements to `thin-edge.io` |
-| `tedge/measurements/<child-id>` | Topic to publish measurements to `thin-edge.io`'s child device |
-| `tedge/errors` | Topic to subscribe to receive any error messages emitted by `thin-edge.io` while processing measurements|
+|Reference           | Description                                                                                                                                                                                |
+| ------------------ |--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|`event-type`        | a string part of the MQTT topic, that identifies the event uniquely in context of the device                                                                                               |
+|`text`              | carries a human readable event-text; must be UTF-8 encoded                                                                                                                                 |
+|`timestamp`         | optional time that indicates when the event has occurred; when not provided, thin-edge.io uses the current system time as the time of the event; when provided must be conform to ISO 8601 |
+|`custom fragments`  | additional fields are handled as custom specific information; if the connected cloud supports custom fragments its mapper transfers those accordingly to the cloud                         |
+
+## Alarms
+
+*Alarms* are notifications about some critical behaviour of the device's environment or software system.
+For instance:
+- a temperature going out of its valid range
+- a process that crashed
+- free disk space going critically low
+
+```sh te2mqtt
+tedge mqtt pub /tedge/alarms/warning/temperature_high '{
+  "text": "Temperature is very high",  // 'text' message of that alarm
+  "time": "2021-01-01T05:30:45+00:00", // optional 'timestamp' of that alarm
+  "someOtherCustomFragment": {         // free 'custom fragments'
+    "nested": {
+      "value": "extra info"
+    }
+  }
+}'
+```
```

Comment:
Removing the json command lines as they can't be used inside a literal string (`''`).

```suggestion
```sh te2mqtt
tedge mqtt pub /tedge/alarms/warning/temperature_high '{
  "text": "Temperature is very high",
  "time": "2021-01-01T05:30:45+00:00",
  "someOtherCustomFragment": {
    "nested": {
      "value": "extra info"
    }
  }
}'
```
```

Reply:
Fixed